### ISSUE TITLE
Suspend Mechanism Switching S0ix/S3 doesn't work.

### ISSUE THREAD

**Issue by johanes2115** [Dasharo/dasharo-issues#466]
**Dasharo version**
v1.4.0

**Dasharo variant**
nv4xmz

**Affected component(s) or functionality**
Suspend

**Brief summary**
S3 (Suspend to RAM) suspend mode is not recognised in the OS when the option is active in BIOS.

**How reproducible**
100%

**How to reproduce**

1. Power on the DUT
2. Enter the BIOS setup using BIOS_SETUP_KEY
3. Choose S3 (Suspend to RAM) option and reboot the DUT

**Expected behavior**
After executing command:
sudo cat /sys/power/mem_sleep
the output should indicate that S3 suspend mode is active.
Expected output:
s2idle [deep]

**Actual behavior**
The output after executing:
sudo cat /sys/power/mem_sleep
indicates that S0iX suspend mode is active:
[s2idle] shallow

**Comment by mkopec** [comment]
@miczyg1 I guess changes to mainboard-specific ASL weren't backported to the TGL models?

**Comment by miczyg1** [comment]
@mkopec yes, we didn't enable S3 support and the switching at all for TGL.

**Comment by macpijan** [comment]
But the report says, the option is there, simply does not work :thinking:

**Comment by miczyg1** [comment]
> But the report says, the option is there in Setup Menu, simply does not work thinking

That's correct.

**Comment by macpijan** [comment]
So we need to either hide it, or implement switching. It certainly can be considered a bug in a current state.

**Comment by philipanda** [comment]
The sleep mode switch is there on `v1.6.0-rc3` S3 is marked experimental.
It seems to not always work, which is a separate thing with the S3 sleep being experimental on some devices.
